### ISSUE TITLE
Rule Change: no-warning-comments detect warning comments in eslint-disable descriptions

### ISSUE THREAD

**Issue by Samuel-Therrien-Beslogic** [eslint/eslint#17887]
### What rule do you want to change?

no-warning-comments

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
// eslint-disable-next-line some-rule-that-needs-fixing -- TODO: Do something

/* eslint-disable-next-line some-rule-that-needs-fixing --
  TODO: Do something,
  but longer to explain */
```

### What does the rule currently do for this code?

Does not trigger

### What will the rule do after it's changed?

Trigger for configured warning comments (here `TODO`) inside eslint disable descriptions

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by Tanujkanti4441** [comment]
Hi @Samuel-Therrien-Beslogic, thanks for the issue but as this rule's docs said 
>This rule reports comments that include any of the predefined terms specified in its configuration.

and eslint directive comments are reserved with some predefined terms and any other terms which are invalid will be reported as an error if we include them with eslint directive comments as you can see here => [playground link](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8td2FybmluZy1jb21tZW50czogXCJlcnJvclwiKi9cblxuLy8gZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIHNvbWUtcnVsZS10aGF0LW5lZWRzLWZpeGluZyAtLSBUT0RPOiBEbyBzb21ldGhpbmdcblxuLyogZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIHNvbWUtcnVsZS10aGF0LW5lZWRzLWZpeGluZyAtLVxuICBUT0RPOiBEbyBzb21ldGhpbmcsXG4gIGJ1dCBsb25nZXIgdG8gZXhwbGFpbiAqLyIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0=)

**Comment by Rec0iL99** [comment]
@Samuel-Therrien-Beslogic could you please provide more info on how you have configured the rule? I guess that you have the default configuration for option `location` in the rule which is `start`. This will only flag comments that start with the warning terms. 

You will have to change it to `anywhere` to make the rule flag warning terms inside ESLint configuration comments.

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXdhcm5pbmctY29tbWVudHM6IFtcImVycm9yXCIsIHsgXCJsb2NhdGlvblwiOiBcImFueXdoZXJlXCIgfV0gKi9cblxuLy8gZXNsaW50LWRpc2FibGUtbmV4dC1saW5lIG5vLXVudXNlZC12YXJzIC0tIFRPRE86IERvIHNvbWV0aGluZ1xuXG4vKiBlc2xpbnQtZGlzYWJsZS1uZXh0LWxpbmUgbm8tdW51c2VkLXZhcnMgLS1cbiAgVE9ETzogRG8gc29tZXRoaW5nLFxuICBidXQgbG9uZ2VyIHRvIGV4cGxhaW4gKi9cbiIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0=)

**Comment by snitin315** [comment]
Yes, it should be `anywhere` for these cases.
